src/ts/jsoneditor.ts

```diff
@@ -68,27 +67,36 @@ $(document).ready(function() {
           }
         }
       });
-      currentFileUploadID = $(this).data('id');
       currentFileItemID = $(this).data('uploadid');
     });
 
-    $(document).on('click', '.jsonSaver', function(){
-      if (typeof currentFileUploadID === 'undefined') {
+    // Clear the JSONEditor and unload the file
+    $(document).on('click', '.jsonClear', function() {
+      currentFileItemID = undefined;
+      editor.set({});
+      $('#jsonEditorTitle').html('File was unloaded.');
+    });
+
+    // The save function is now defined separately
+    const saveJsonFile = function(){
+      if (typeof currentFileItemID === 'undefined') {
         // we are creating a new file
         const realName = prompt('Enter name of the file');
         if (realName == null) {
           return;
         }
-        const id = $('#main_form').find('input[name="id"]').attr('value');
+        // add the new name for the file as a title
+        $('#jsonEditorTitle').html('Filename: ' + realName + '.json');
         $.post('app/controllers/EntityAjaxController.php', {
           addFromString: true,
           type: 'experiments',
-          id: id,
+          id: $('#info').data('id'),
```

Comment:
I wish to deprecate this `#info` div. So you should add a `data-id` on the element.

Reply:
That's fixable. But whats the new alternative for:  

- `if (!($('#info').data('page') === 'edit' || $('#info').data('page') === 'view'))`
- `modes: (($('#info').data('page') === 'edit') ? ['tree','code','view','form','text']:['view'])`

Reply:
it would be:

~~~js
const params = new URLSearchParams(document.location.search);
console.log('mode = ' + params.get('mode'));
~~~

But don't worry too much about it.

Reply:
I have left it for now then.